test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -87,6 +64,41 @@ abstract class DirectMethodSenderBase : IDisposable
             }
         }
 
+        async Task<int> InvokeDirectMethodWithRetryAsync(
+            ILogger logger,
+            string deviceId,
+            string targetModuleId,
+            string methodName,
+            ulong directMethodCount)
+        {
+            const int maxRetry = 3;
+            int resultStatus = (int)HttpStatusCode.InternalServerError;
+            int transientRetryCount = 0;
+
+            while (transientRetryCount <= maxRetry)
+            {
+                try
+                {
+                    transientRetryCount++;
+                    resultStatus = await this.InvokeDeviceMethodAsync(deviceId, targetModuleId, methodName, directMethodCount, CancellationToken.None);
+                }
+                catch (IotHubCommunicationException e) when (e.IsTransient)
+                {
+                    if (transientRetryCount < maxRetry)
```

Comment:
Remove this if block and just keep line 89 and 94; and add transientRetryCount into log message in line 89.

Reply:
OOOH I see what you mean. nice!